industry_benchmarks/utils/plan_rbfe_network.py

```diff
@@ -0,0 +1,227 @@
+    settings.alchemical_settings.explicit_charge_correction = True
+    settings.simulation_settings.production_length = 20 * unit.nanosecond
+    settings.simulation_settings.n_replicas = 22
+    settings.lambda_settings.lambda_windows = 22
+    return settings
+
+
+@click.command
+@click.option(
+    '--ligands',
+    type=click.Path(dir_okay=False, file_okay=True, path_type=pathlib.Path),
+    required=True,
+    help="Path to the prepared SDF file containing the ligands",
+)
+@click.option(
+    '--pdb',
+    type=click.Path(dir_okay=False, file_okay=True, path_type=pathlib.Path),
+    required=True,
+    help="Path to the prepared PDB file of the protein",
+)
+@click.option(
+    '--cofactors',
+    type=click.Path(dir_okay=False, file_okay=True, path_type=pathlib.Path),
+    default=None,
+    help="Path to the prepared cofactors SDF file (optional)",
+)
+@click.option(
+    '--output',
+    type=click.Path(dir_okay=True, file_okay=False, path_type=pathlib.Path),
+    default=None,
+    help="Directory name in which to store the transformation json files",
+)
+def run_inputs(ligands, pdb, cofactors, output):
+    """
+    Generate run json files for RBFE calculations
+
+    Parameters
+    ----------
+    ligands : pathlib.Path
+      A Path to a ligands SDF.
+    pdb : pathlib.Path
+      A Path to a protein PDB file.
+    cofactors : Optional[pathlib.Path]
+      A Path to an SDF file containing the system's cofactors.
+    output: pathlib.Path
+      A Path to a directory where the transformation json files
+      will be stored into.
+    """
+    # Create the output directory
+    if not os.path.isdir(output):
+        os.mkdir(output)
+    # Create the small molecule components of the ligands
+    rdmols = [mol for mol in Chem.SDMolSupplier(str(ligands), removeHs=False)]
+    smcs = [openfe.SmallMoleculeComponent.from_rdkit(mol) for mol in rdmols]
+    # Generate the partial charges
+    logger.info("Generating partial charges for ligands")
+    smcs = [gen_charges(smc) for smc in smcs]
+
+    # Create ligand network
+    ligand_network = gen_ligand_network(smcs)
+
+    # Create the solvent and protein components
+    solv = openfe.SolventComponent()
+    prot = openfe.ProteinComponent.from_pdb_file(str(pdb))
+
+    # If we have cofactors, load them in and assign partial charges
+    if cofactors is not None:
+        cofactors_smc = [gen_charges(openfe.SmallMoleculeComponent(m))
+                         for m in Chem.SDMolSupplier(str(cofactors), removeHs=False)]
+
+    # Create the AlchemicalTransformations, and storing them to disk
+    for mapping in ligand_network.edges:
+        # Get different settings depending on whether the transformation
+        # involves a change in net charge
+        charge_difference = get_alchemical_charge_difference(mapping)
+        if abs(charge_difference) > 1e-3:
+            rfe_settings = get_settings_charge_changes()
```

Comment:
I'd raise a warning to users when this happens so that they know that there's a charge change